Packages/com.unity.inputsystem/InputSystem/Plugins/Linux/SDLDeviceBuilder.cs

```diff
@@ -0,0 +1,329 @@
+                    sanitizedName.Append(letter);
+                }
+            }
+            return sanitizedName.ToString();
+        }
+
+        internal static string OnFindLayoutForDevice(int deviceId, ref InputDeviceDescription description, string matchedLayout, IInputRuntime runtime)
+        {
+            // If the device isn't a XRInput, we're not interested.
+            if (description.interfaceName != SDLSupport.kXRInterfaceCurrent)
+            {
+                return null;
+            }
+
+            // If the description doesn't come with a XR SDK descriptor, we're not
+            // interested either.
+            if (string.IsNullOrEmpty(description.capabilities))
+            {
+                return null;
+            }
+
+            // Try to parse the SDL descriptor.
+            SDLDeviceDescriptor deviceDescriptor;
+            try
+            {
+                deviceDescriptor = SDLDeviceDescriptor.FromJson(description.capabilities);
+            }
+            catch (Exception)
+            {
+                return null;
+            }
+
+            if (deviceDescriptor == null)
+            {
+                return null;
+            }
+
+            if (string.IsNullOrEmpty(matchedLayout))
+            {
+                //matchedLayout = "Joystick";
+            }
+
+            string layoutName = null;
+            if (string.IsNullOrEmpty(description.manufacturer))
+            {
+                layoutName = string.Format("{0}::{1}", SanitizeName(description.interfaceName),
+                    SanitizeName(description.product));
+            }
+            else
+            {
+                layoutName = string.Format("{0}::{1}::{2}", SanitizeName(description.interfaceName), SanitizeName(description.manufacturer), SanitizeName(description.product));
+            }
+
+            var layout = new SDLLayoutBuilder { descriptor = deviceDescriptor, parentLayout = matchedLayout };
+            InputSystem.RegisterLayoutBuilder(() => layout.Build(), layoutName, matchedLayout);
+
+            return layoutName;
+        }
+
+        internal bool IsAxisX(SDLFeatureDescriptor feature)
+        {
+            return feature.featureType == JoystickFeatureType.Axis
+                && feature.usageHint == (int)SDLAxisUsage.X;
+        }
+
+        internal bool IsAxisY(SDLFeatureDescriptor feature)
+        {
+            return feature.featureType == JoystickFeatureType.Axis
+                && feature.usageHint == (int)SDLAxisUsage.Y;
+        }
+
+        internal void BuildStickFeature(ref InputControlLayout.Builder builder, SDLFeatureDescriptor xFeature, SDLFeatureDescriptor yFeature)
+        {
+            int byteOffset;
+            if (xFeature.offset <= yFeature.offset)
+            {
+                byteOffset = xFeature.offset;
+            }
+            else
+            {
+                byteOffset = yFeature.offset;
+            }
+
+            var stickName = "Stick";
```

Comment:
What if there is more than one stick?

Reply:
Wondering in general, is every axis and button usage guaranteed to be unique? If so, that point is moot.